### PR TITLE
Creating web api middleware and logging

### PR DESCRIPTION
**Pull request by AntonBakum**

### CODE REVIEW

File: pet-api/pet-api/Infrastructure/Logging/FileLogger.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿namespace pet_api.Infrastructure.Logging
+{
+    public class FileLogger : ILogger
+    {
+        string path;
```
Comment (YevheniiDomnich) [inline pet-api/pet-api/Infrastructure/Logging/FileLogger.cs]: add private readonly. Always write the modifiers explicitly, it's a good practice

File: pet-api/pet-api/Infrastructure/Logging/FileLoggerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace pet_api.Infrastructure.Logging
+{
+    public static class FileLoggerExtensions
+    {
+        public static string CreateFile()
```
Comment (YevheniiDomnich) [inline pet-api/pet-api/Infrastructure/Logging/FileLoggerExtensions.cs]: Is this method used outside this class?

File: pet-api/pet-api/Infrastructure/Logging/FileLoggerProvider.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Concurrent;
+
+namespace pet_api.Infrastructure.Logging
+{
+    public class FileLoggerProvider : ILoggerProvider
+    {
+        string path;
```
Comment (YevheniiDomnich) [inline pet-api/pet-api/Infrastructure/Logging/FileLoggerProvider.cs]: private readonly

File: pet-api/pet-api/Controllers/WeatherForecastController.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherForecastController : ControllerBase
+    {
+        private static readonly string[] Summaries = new[]
+        {
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+    };
+
+        private readonly ILogger<WeatherForecastController> _logger;
```
Comment (YevheniiDomnich) [inline pet-api/pet-api/Controllers/WeatherForecastController.cs]: I don't see where this _logger is used. Is it useless here?

File: pet-api/pet-api/Infrastructure/Logging/FileLoggerExtensions.cs (hunk 2 above)
Comment (YevheniiDomnich) [reply pet-api/pet-api/Infrastructure/Logging/FileLoggerExtensions.cs]: and it seems like the name is incorrect. It creates the path, am I right?

File: pet-api/pet-api/Infrastructure/Logging/FileLogger.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            return logLevel != LogLevel.None;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+            using (FileStream stream = File.Open(path, FileMode.Append))
```
Comment (YevheniiDomnich) [inline pet-api/pet-api/Infrastructure/Logging/FileLogger.cs]: Good solution using the FileMode.Append parameter! Approved.

Comment (YevheniiDomnich) [reply pet-api/pet-api/Infrastructure/Logging/FileLogger.cs]: I'm afraid that this approach is not thread-safe, or I missed something and we can work with this file via >2 threads in parallel? If not - please, add some synchronization mechanism.

File: pet-api/pet-api/Infrastructure/Logging/FileLogger.cs (hunk 1 above)
Comment (AntonBakum) [reply pet-api/pet-api/Infrastructure/Logging/FileLogger.cs]: modifier has been added

File: pet-api/pet-api/Infrastructure/Logging/FileLoggerExtensions.cs (hunk 2 above)
Comment (AntonBakum) [reply pet-api/pet-api/Infrastructure/Logging/FileLoggerExtensions.cs]: New  access modifier: "private", new name: "CreatePath"

File: pet-api/pet-api/Infrastructure/Logging/FileLoggerProvider.cs (hunk 3 above)
Comment (AntonBakum) [reply pet-api/pet-api/Infrastructure/Logging/FileLoggerProvider.cs]: The modifier has been added

File: pet-api/pet-api/Controllers/WeatherForecastController.cs (hunk 4 above)
Comment (AntonBakum) [reply pet-api/pet-api/Controllers/WeatherForecastController.cs]: I have been deleted this useless logger from the controller

File: pet-api/pet-api/Infrastructure/Logging/FileLogger.cs (hunk 5 above)
Comment (AntonBakum) [reply pet-api/pet-api/Infrastructure/Logging/FileLogger.cs]: lock (_lock) construction has been added to provide the thread-safety

File: pet-api/pet-api/Infrastructure/Logging/FileLogger.cs (hunk 1 above)
Comment (YevheniiDomnich) [reply pet-api/pet-api/Infrastructure/Logging/FileLogger.cs]: Resolved

File: pet-api/pet-api/Program.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+using pet_api.Infrastructure.Logging;
+using pet_api.Middleware.ErrorHandlingMiddleware;
+using pet_api.Middleware.LoggingMiddleware;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddLogging(loggingBuilder =>
+{
+    loggingBuilder.ClearProviders();
+    loggingBuilder.AddFile();
```
Comment (YevheniiDomnich) [inline pet-api/pet-api/Program.cs]: Maybe rename this method name to Add FileLoggerProvider? I think it will be more clear.

**Pull request merged**